src/halmos/sevm.py

```diff
@@ -1733,12 +1737,8 @@ def bitwise(op, x: Word, y: Word) -> Word:
             return x ^ y  # bvxor
         else:
             raise ValueError(op, x, y)
-    elif is_bool(x) and is_bv(y):
-        return bitwise(op, If(x, ONE, ZERO), y)
-    elif is_bv(x) and is_bool(y):
-        return bitwise(op, x, If(y, ONE, ZERO))
     else:
-        raise ValueError(op, x, y)
+        return bitwise(op, b2i(x), b2i(y))
```

Comment:
this would result in infinite recursion if x or y is neither bool or bv.  it doesn't seem to happen in the current code, but it might be in the future.  adding an explicit check would be safer.

Reply:
that's good idea too!